README.md
Deployed Link:
https://bffhackathon-final-7a7h.vercel.app/

Youtube Explaination Video Link:
https://youtu.be/qaTQMl4UvRs

Introduction
 It suggests a platform for gamers to come together and forge new connections and opportunities in the gaming world. The name also implies a sense of strength and community, which could be appealing to users looking to build their profiles, compete in tournaments, and seek out sponsorships.
 
 HOW IT WORKS
It lists all the gaming tournaments to take place.
It helps in maintaining a gaming profile for gamers during their E-Sports tenure and updates their perfpormance based on their tournament results including their gaming rating and special skills.
It helps the organizers to build a great gaming team based on the gaming profiles built on our platform.

 The problem it solves
It solves four major problems in the field of Esports 
•	No pathway for the beginners in competitive gaming community to reach upper levels
•	Its hard for organizations to find good players for their gaming lineup as gamers do not have a proper portfolio
•	Its hard for players to find sponsors for themselves 
•	Game making Companies do not have proper alpha and beta testers among the gamers


FEATURES
A single site to list all gaming tournaments.
Maintaining a gaming profile for gamers for a long period of time.
Helps sponsors to select gamers based on their gaming profiles.

Challenges you run into while building this project
The USP of our project is the gamers profile and its statistics needed a lot of pie charts and graphs to be used and proper integration which was initially giving a lot of errors and bugs so we did take help from online tech stack documentation and after working a lot on it finally fixed the bugs and made the website


Technologies we used
HTML
CSS
Javascript
ReactJS

PRSENTATION LINK:
https://www.canva.com/design/DAFe5Pt9Uwc/jwvq5-mGipJaxEVnpSXXSg/view?utm_content=DAFe5Pt9Uwc&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton


CHANGELOG.md
# CHANGELOG

## [1.2.1] 2022-06-29

- Update dependencies
- Migrate from node-sass to sass

## [1.2.0] 2020-12-11

### Bug fixing

- Added hooks support
- https://github.com/creativetimofficial/black-dashboard-react/issues/19
- https://github.com/creativetimofficial/black-dashboard-react/issues/17

### Major style changes

- All Bootstrap imports from `src/assets/scss/black-dashboard-react/bootstrap` have been changed with imports from `node_modules` from `Bootstrap`

### Deleted components

### Added components

### Deleted dependencies

- history (instead of using `history`, we'll be using `BrowserRouter` from `react-router-dom`)
- @types/googlemaps
- @types/react
- @types/markerclustererplus
- react-google-maps (Instead of it, we're going to use simple plain JS Google maps)

### Added dependencies

- node-sass-package-importer@5.3.2 (for importing `Bootstrap` from `node_modules`)
- bootstrap@4.5.3 (instead of using downloaded zip of `Bootstrap`, we'll use it from `node_modules`)
- jquery@3.5.1 (to stop `Bootstrap` warning on a clean install)

### Updated dependencies

```
@fortawesome/fontawesome-free    5.13.0   →   5.15.1
chart.js                          2.9.3   →    2.9.4
node-sass                        4.13.1   →   4.14.1
react                           16.13.1   →   17.0.1
react-chartjs-2                   2.9.0   →   2.11.1
react-dom                       16.13.1   →   17.0.1
react-notification-alert         0.0.12   →   0.0.13
react-router-dom                  5.1.2   →    5.2.0
react-scripts                     3.4.1   →    4.0.1
reactstrap                        8.4.1   →    8.7.1
eslint-plugin-flowtype           3.13.0   →    5.2.0
typescript                        3.8.3   →    4.1.2
```

### Important Notes

**The jQuery and TypeScript dependencies are installed only to stop console warnings on install. They are not actually used in our product. So the product is not based on jQuery, and it is not based on TypeScript!**

### Warning

_Some warnings may appear when running the installation command, but they do not affect the UI or the functionality of the product._
_The following warnings will appear when running the installation command, but they do not affect the UI or the functionality of the product (they will be solved in our next update - they come from the plugins that we are using, and they haven't yet upgraded to the latest React version):_

```
npm WARN react-popper@1.3.7 requires a peer of react@0.14.x || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN create-react-context@0.3.0 requires a peer of react@^0.14.0 || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
```

Happy Hacking!

## [1.1.0] 2020-03-13

### Bug fixing

- Github issues:
  - https://github.com/creativetimofficial/black-dashboard-react/issues/8
  - https://github.com/creativetimofficial/black-dashboard-react/issues/7
  - https://github.com/creativetimofficial/black-dashboard-react/issues/6 (not really an issue, but was old and needed to close it)
  - https://github.com/creativetimofficial/black-dashboard-react/issues/4

### Major style changes

### Deleted components

### Added components

### Deleted dependencies

- bootstrap (styles are already added inside the scss folder)

### Added dependencies

- To stop consolde warnings on installation:
  - eslint-plugin-flowtype@3.13.0
  - @types/markerclustererplus@2.1.33
  - @types/googlemaps@3.39.3
  - typescript@3.8.3
- For GAMEFORGE copyrights
  - gulp@4.0.2
  - gulp-append-prepend@1.0.8

### Updated dependencies

```
@types/react               16.4.16   →   16.9.23
chart.js                    ^2.7.2   →     2.9.3
history                      4.7.2   →    4.10.1
node-sass                    4.9.3   →    4.13.1
perfect-scrollbar            1.4.0   →     1.5.0
prop-types                 ^15.6.2   →    15.7.2
react                       16.5.2   →   16.13.0
react-chartjs-2             ^2.7.4   →     2.9.0
react-dom                   16.5.2   →   16.13.0
react-notification-alert     0.0.8   →    0.0.12
react-router-dom             4.3.1   →     5.1.2
react-scripts                2.0.4   →     3.4.0
reactstrap                   6.5.0   →     8.4.1
```

### Warning

**There are some errors that come from the react-google-maps components, since they are using old React v16 syntax, this, at the moment does not affect the UI or the functionality of the product. If the issue will perssit in React v17, we will drop the usage of these components and replace them with other maps components. The following warnings on installation could not be solved, due to some of our dependencies, however, they do not affect the functionality or the UI of the product:**

```
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated popper.js@1.16.1: Popper changed home, find its new releases at @popperjs/core
```

**Also, the above, are only present in development, and not in production.**

## [1.0.0] 2018-11-29

### Original Release

- Added Reactstrap as base framework
- Added design from PRODUCT NAME by GAMEFORGE


src/views/Dashboard.js


import React from "react";
// nodejs library that concatenates classes
import classNames from "classnames";
// react plugin used to create charts
import { Line, Bar } from "react-chartjs-2";
// import Card from 'react-bootstrap/Card';

// reactstrap components
import {
  Button,
  ButtonGroup,
  Card,
  CardHeader,
  CardBody,
  CardTitle,
 
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  UncontrolledDropdown,
  Label,
  FormGroup,
  Input,
  Table,
  Row,
  Col,
  UncontrolledTooltip
} from "reactstrap";

// core components
import {
  chartExample1,
  chartExample2,
  chartExample3,
  chartExample4
} from "variables/charts.js";

function Dashboard(props) {
  const [bigChartData, setbigChartData] = React.useState("data1");
  const setBgChartData = (name) => {
    setbigChartData(name);
  };
  return (
    <>
      <div className="content">
        <Row>
          <Col xs="12">
            <Card className="card-chart">
              <CardHeader>
                <Row>
                  <Col className="text-left" sm="6">
                    <h5 className="card-category">Rating Vs Time</h5>
                    <CardTitle tag="h2">1010 Rating</CardTitle>
                  </Col>
                  <Col sm="6">
                    <ButtonGroup
                      className="btn-group-toggle float-right"
                      data-toggle="buttons"
                    >
                      <Button
                        tag="label"
                        className={classNames("btn-simple", {
                          active: bigChartData === "data1"
                        })}
                        color="info"
                        id="0"
                        size="sm"
                        onClick={() => setBgChartData("data1")}
                      >
                        <span className="d-none d-sm-block d-md-block d-lg-block d-xl-block">
                          Month-Wise
                        </span>
                        <span className="d-block d-sm-none">
                          <i className="tim-icons icon-single-02" />
                        </span>
                      </Button>
                      <Button
                        color="info"
                        id="1"
                        size="sm"
                        tag="label"
                        className={classNames("btn-simple", {
                          active: bigChartData === "data2"
                        })}
                        onClick={() => setBgChartData("data2")}
                      >
                        <span className="d-none d-sm-block d-md-block d-lg-block d-xl-block">
                          Year-Wise
                        </span>
                        <span className="d-block d-sm-none">
                          <i className="tim-icons icon-gift-2" />
                        </span>
                      </Button>
                     
                    </ButtonGroup>
                  </Col>
                </Row>
              </CardHeader>
              <CardBody>
                <div className="chart-area">
                  <Line
                    data={chartExample1[bigChartData]}
                    options={chartExample1.options}
                  />
                </div>
              </CardBody>
            </Card>
          </Col>
        </Row>
        <Row>
          <Col lg="4">
          <Card className="card-chart">
          {/* <Card.Img variant="top" className='imgs' src="https://resources.esportsinsider.com/esportsinsider/2020/06/Ecricket-World-Series.jpg" /> */}
              <CardHeader>
              
                <h3 className="card-category"><h2>Tournaments Status</h2></h3>
                <CardTitle tag="h3">
                  {/* <i className="tim-icons icon-send text-success" /> */}
                </CardTitle>
              </CardHeader>
              <CardBody>
               
                <img src='https://lh3.googleusercontent.com/g0Jw-I6-gH2DVCpnl3u8QKZVT_meR9lcJlpyeSZ-MyvwLnyEZvgyrY5frldA8HCv55s=w280-rwa' height='200px' width='250px'></img>
                
              </CardBody>
            </Card>

           
          </Col>
          <Col lg="4">
          <Card className="card-chart">
          
              <CardHeader>
                <h5 className="card-category"><h2>Your Starred Rating</h2> </h5>
                <CardTitle tag="h3">
                  {/* <i className="AiFillStar icon-send text-success" /> 12,100K */}
                </CardTitle>
              </CardHeader>
              <CardBody>
                <img src="https://www.shutterstock.com/shutterstock/videos/1095126757/thumb/1.jpg?ip=x480" />
                
                
              </CardBody>
            </Card>
            
          </Col>
          <Col lg="4">
            <Card className="card-chart">
              <CardHeader>
                
                <CardTitle tag="h3">
                <h2>Special Skills</h2>
                  
                </CardTitle>
              </CardHeader>
              <CardBody>
                <ul type='none'>
                  <h3>
                  <li>In-Game-Leadering</li>
                  <li>Snipping</li>
                  <li>Good in attacking</li>
                  </h3>
                  
                </ul>
                
              </CardBody>
            </Card>
          </Col>
        </Row>
        <Row>
          <Col lg="3" md="12">
           
          </Col>
          <Col lg="6" md="12">
            <Card>
              <CardHeader>
                <CardTitle tag="h4">Past Tornaments</CardTitle>
              </CardHeader>
              <CardBody>
                <Table className="tablesorter" responsive>
                  <thead className="text-primary">
                    <tr>
                      <th>Tournament Name</th>
                      <th>Organizer</th>
                      <th>Date & time</th>
                      <th className="text-center">Status</th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>Dakota Rice</td>
                      <td>Sky E-sports</td>
                      <td>2023-03-29</td>
                      <td className="text-center win">₹10000 Won</td>
                    </tr>
                    <tr>
                      <td>Minerva Hooper</td>
                      <td>VLT</td>
                      <td>2023-03-20</td>
                      <td className="text-center win">₹70000 Won</td>
                    </tr>
                    <tr>
                      <td>Sage Rodriguez</td>
                      <td>S8UL</td>
                      <td>2023-03-15</td>
                      <td className="text-center win">₹10000</td>
                    </tr>
                    <tr>
                      <td>Philip Chaney</td>
                      <td>VLT</td>
                      <td>2023-03-08</td>
                      <td className="text-center lost">Lost</td>
                    </tr>
                    <tr>
                      <td>Doris Greene</td>
                      <td>S8UL</td>
                      <td>2023-02-21</td>
                      <td className="text-center lost">Lost</td>
                    </tr>
                    <tr>
                      <td>Mason Porter</td>
                      <td>Sky E-sports</td>
                      <td>2023-02-17</td>
                      <td className="text-center win">₹120000 Won</td>
                    </tr>
                    <tr>
                      <td>Jon Porter</td>
                      <td>S8UL</td>
                      <td>2023-02-05</td>
                      <td className="text-center win">₹15000 Won</td>
                    </tr>
                  </tbody>
                </Table>
              </CardBody>
            </Card>
          </Col>
        </Row>
      </div>
    </>
  );
}

export default Dashboard;
